Meta/build-root-filesystem.sh

```diff
@@ -46,14 +46,14 @@ LLVM_VERSION="${LLVM_VERSION:-13.0.0}"
 
 if [ "$SERENITY_TOOLCHAIN" = "Clang" ]; then
     TOOLCHAIN_DIR="$SERENITY_SOURCE_DIR"/Toolchain/Local/clang/
-    $CP "$TOOLCHAIN_DIR"/lib/"$SERENITY_ARCH"-pc-serenity/* mnt/usr/lib
+    $CP --preserve=timestamps "$TOOLCHAIN_DIR"/lib/"$SERENITY_ARCH"-pc-serenity/* mnt/usr/lib
```

Comment:
Do FreeBSD/OpenBSD/MacOS support this? We should try to avoid breaking the other hosts we attempt to support.

Reply:
https://superuser.com/questions/114195/how-can-i-copy-a-file-in-unix-without-altering-its-last-modified-time the man pages listed here seem to suggest the `--preserve=` version might be specific to Linux?

Reply:
I assumed it would be covered by the GNU coreutils fallback at the top of the page:

```
CP="cp"

# cp on macOS and BSD systems do not support the -d option.
# gcp comes with coreutils, which is already a dependency.
OS="$(uname -s)"
if [ "$OS" = "Darwin" ] || echo "$OS" | grep -qe 'BSD$'; then
	CP="gcp"
fi
```

The option `--preserve` with custom selectors definitely shows up for `gcp` on macOS.

Reply:
I see, maybe we should update that comment about the `-d option` to also include the fact that now depend on it for the `--preserve` option?

Reply:
Done. (Although, I removed the reference to the `-d` option from the comment entirely as it was no longer in use.)